sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/SelectQuery.cs

```diff
@@ -0,0 +1,103 @@
+        internal SelectQuery(AdtQueryBuilder parent, FromQuery upstreamFromQuery)
+        {
+            _parent = parent;
+            _delegationClause = upstreamFromQuery;
+        }
+
+        /// <summary>
+        /// Used to add a select clause (and its corresponding argument) to the query.
+        /// </summary>
+        /// <param name="args"> The arguments that define what we select (eg. *). </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery Select(params string[] args)
+        {
+            _clause = new SelectClause(args);
+            return _delegationClause;
+        }
+
+        /// <summary>
+        /// Used when applying the TOP() aggregate from the ADT query language. Same functionality as select
+        /// but inserts TOP() into the query structure as well.
+        /// </summary>
+        /// <param name="count"> The argument for TOP(), ie the number of instances to return. </param>
+        /// <param name="args"> The arguments that can be optionally passed with top (eg property name). </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectTop(int count, params string[] args)
+        {
+            // turn into correct format -- eg. SELECT TOP(3)
+            StringBuilder topArg = new StringBuilder().Append($"{QueryConstants.Top}({count})").Append(' ');
+
+            // append optional arguments separated by commas
+            topArg.Append(string.Join(", ", args));
+
+            _clause = new SelectClause(new string[] { topArg.ToString() });
+            return _delegationClause;
+        }
+
+        /// <summary>
+        /// Used when applying the COUNT() aggregate from the ADT query language. Similar to SelectTop().
+        /// </summary>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectCount()
+        {
+            string countArg = new StringBuilder().Append($"{QueryConstants.Count}()").Append(' ').ToString();
+
+            _clause = new SelectClause(new string[] { countArg });
+            return _delegationClause;
+        }
+
+        /// <summary>
+        /// Used when overriding the query builder with the literal query string.
+        /// </summary>
+        /// <param name="literalQuery"> Query in string format. </param>
+        /// <returns> Query that contains a select clause. </returns>
+        public FromQuery SelectOverride(string literalQuery)
+        {
+            // TODO -- just use select?
+            // approach 1 - method overload
+            // approach 2 - seperate method
+            _clause = new SelectClause(new string[] { literalQuery });
+
+            return _delegationClause;
+        }
+
+        /// <inheritdoc/>
+        public override AdtQueryBuilder Build()
+        {
+            throw new InvalidOperationException("Queries that contain just a SELECT statement are not valid.");
```

Comment:
This must be a TODO eh?

Reply:
Well, you can't just right a query (at least not yet) that just goes something like `SELECT *` -- you need to specify a collection so the minimum would be something like `SELECT * FROM DIGITALTWINS` -- we can ask the ADT query lang expert on their opinion about this though to see if that's something that could be useful.

Reply:
Oh, I think I understand you now. What you're saying is that `Build()` on just select query is invalid, so the method on this class should never be called. Is that right?

Reply:
I think the statement can be improved upon by mentioning that the query is missing a FROM statement. Just saying that select by itself is not valid is good, but doesn't provide additional information as to how do I fix that ...